### PR TITLE
Add melt method

### LINKED ISSUE #59: Pandas melt() alternative?
I'm trying to find how to do something like the pandas melt() method.

Basically I need to go from this:

```
|Date|A|B|C|D|E |
|----|-|-|-|-|--|
|2005|1|2|3|4|50|
|2006|6|7|8|9|10|
```

Doing something like:
`df.melt("Date", { varName: "X", valueName: "Val" })`

And get this:
```
|Date|X|Val|
|----|-|---|
|2005|A|1  |
|2005|B|2  |
|2005|C|3  |
etc 
|2006|A|6  |
etc
|2006|E|10 |
```

### PR DESCRIPTION
**Pull request by karishnu**
fix #59

### CODE REVIEW

File: src/test/iterators/tile-iterator.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import { expect } from 'chai';
+import 'mocha';
+import { TileIterator } from '../../lib/iterators/tile-iterator';
+import { ArrayIterable } from '../../lib/iterables/array-iterable';
+
+describe('repeat iterator', function () {
```
[the remark this answers is not in the document]
Comment (karishnu) [reply src/test/iterators/tile-iterator.test.ts:6]: Yes, it should be "tile iterator"! Don't know how I missed that one. Will change it.

File: src/lib/iterators/ravel-iterator.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+//
+// An iterator that produces a contiguous flattened array generated from each set of elements in child iterables.
+// Implementation similar to https://docs.scipy.org/doc/numpy/reference/generated/numpy.ravel.html with order 'C'
+//
+// Note that numpy does not work with arrays of varying lengths. However, the dataforge implementation ignores
+// that requirement due to the iterator architecture. Moreover, since the RavelIterator is only used internally
+// such cases will not occur.
+//
+
+export class RavelIterator<T> implements Iterator<T> {
```
[the remark this answers is not in the document]
Comment (karishnu) [reply src/lib/iterators/ravel-iterator.ts:10]: It performs the same function as the Ravel method in Numpy (https://numpy.org/doc/stable/reference/generated/numpy.ravel.html). The ravel method was used in the Pandas implementation of melt.

File: src/test/iterators/tile-iterator.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (karishnu) [reply src/test/iterators/tile-iterator.test.ts]: I have updated the PR

[the remark this answers is not in the document]
Comment (ashleydavis) [reply src/test/iterators/tile-iterator.test.ts]: Thanks

File: src/lib/iterators/ravel-iterator.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashleydavis) [reply src/lib/iterators/ravel-iterator.ts:10]: Wow that's so interesting. I looked up ravel and the word means "put together". It's not often I hear a new word!

**Comment by ashleydavis** [comment]
Merged, thanks for you contribution. I'll release a new version soon.

**Comment by ashleydavis** [comment]
I've published this contribution in version 1.8.9.

**Pull request merged**